File: js/lv1/정수제곱근판별.js
function solution(n) {
  const sqrt_n = parseInt(n ** (1 / 2));
  return n === sqrt_n ** 2 ? (sqrt_n + 1) ** 2 : -1;
}

N = [121, 3];
answer = [144, -1];

for (let i = 0; i < N.length; i++) {
  if (solution(N[i]) == answer[i]) {
    console.log(`${i + 1}: pass`);
  } else {
    console.log(`${i + 1}: fail`);
  }
}

// function nextSqaure(n) {
//   var result = 0;
//   var n = Math.sqrt(n);
//   result = Number.isInteger(n) ? Math.pow(n + 1, 2) : "no";
//   return result;
// }

File: python/lv1/정수제곱근판별.py
from math import sqrt

def solution(n):
    if sqrt(n) == int(sqrt(n)):
        return (sqrt(n) + 1) ** 2
    else:
        return -1

N = [121, 3]
answer = [144, -1]

for i in range(len(N)):
    if solution(N[i]) == answer[i]:
        print(f'{i+1}: pass')
    else:
        print(f'{i+1}: fail')

# def nextSqure(n):
#     sqrt = n ** (1/2)

#     if sqrt % 1 == 0:
#         return (sqrt + 1) ** 2
#     return 'no'

# def nextSqure(n):
#     return n == int(n**.5)**2 and int(n**.5+1)**2 or 'no'